### PR TITLE
Fixed #33111 -- Fixed passing object to ModelAdmin.get_inlines() when editing in admin change view.

### PR DESCRIPTION
**Pull request by hramezani**
https://code.djangoproject.com/ticket/33111

### CODE REVIEW

**Comment by kezabelle** [comment]
This feels like it _might_ need a docs note somewhere, because users _may_ be reliant on the old behaviour, where the formsets are created based on the _mutated, validated_ parent Model object I think? Subsequent to this change, everything downstream of the `_create_formsets` method will receive the unmutated ("before-save") parent Model instance (which itself should still technically have been previously _validated_ to get in the DB, but not guaranteed).

As far as I can tell from the down-stream methods which are public (`get_inline_instances`, `get_formsets_with_inlines` etc) none specifically say what state the object being edited is in though, so maybe not?

File: tests/admin_inlines/models.py [hunk 1]
```diff
@@ -326,6 +326,15 @@ class Meta:
         proxy = True
 
 
+# Models for #33111
```
Comment (felixxm) [inline tests/admin_inlines/models.py:329]: This comment is unnecessary.

File: tests/admin_inlines/admin.py [hunk 2]
```diff
@@ -308,6 +309,18 @@ class ClassStackedHorizontal(admin.StackedInline):
     filter_horizontal = ['person']
 
 
+# Admins for #33111
```
Comment (felixxm) [inline tests/admin_inlines/admin.py:312]: This comment is unnecessary.

**Review by felixxm — COMMENTED** [review]
@hramezani Thanks for this patch :+1: 

> As far as I can tell from the down-stream methods which are public (`get_inline_instances`, `get_formsets_with_inlines` etc) none specifically say what state the object being edited is in though, so maybe not?

There is no need for a note, IMO :thinking: . Formsets are rendered based on the `obj` previous state so we should create them with the same.

**Comment by kezabelle** [comment]
> There is no need for a note, IMO. Formsets are rendered based on the `obj` previous state so we should create them with the same.

Fair enough, I was just throwing it out there as a thing to note. 

FWIW, It's not the _rendering_  state that is being changed, though, it is the _validation_ state. Previously, creating formsets during a POST request happened _after_ the parent modelform has been validated, which meant the `obj` given to `_create_formsets` (and the downstream public methods thereof) was the mutated, validated instance. Now it'll be the unmutated, unvalidated instance. Naturally that should itself be valid from previously being saved, but the mutations to it during this request (eg: any custom `clean_x` to massage the value) won't yet have been applied.

I sincerely doubt anyone's _relying_ on the behaviour, though.

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Pull request merged**